src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -270,6 +270,9 @@ public function get($id, $invalidBehavior = self::EXCEPTION_ON_INVALID_REFERENCE
 
                     $alternatives = array();
                     foreach ($this->services as $key => $associatedService) {
+                        if (isset($this->privates[$key])) {
```

Comment:
> Not sure why we want $private = true. Opposed to refactoring stuff relying on the fact it returns private ids.

That's only in order to ensure BC. Having this argument set to true should trigger a deprecation. Internally, we'll only use `false`. (if any internal call needs privates services, we'll have to add another `$triggerDeprecation = true` argument and set it to `false`, but I doubt there is.)